flink-libraries/flink-table/src/test/scala/org/apache/flink/api/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -1101,6 +1101,45 @@ class ScalarFunctionsTest extends ExpressionTestBase {
       "true")
   }
 
+  @Test
+  def testInExpressions(): Unit = {
+    testTableApi(
```

Comment:
@twalthr It's a big problem. Even two problems:
1. We mention that that all types (right) in IN operator have the same type because we use Set as collection. And we check if right operand has only one `TypeInfo`. Otherwise we should check each item one by one or use something like bloom filter
2. Problem with compilation of expression - each field is compiling to more simple type (the most big problem)